### DISCUSSION TITLE
custom build / radio not found

### DISCUSSION THREAD

**Discussion by btalbot85** [markqvist/Reticulum#494] [General]
Well I am so close and I think I have done everything correctly so far??

I have compiled a custom board, set the product, model, etc and it compiles fine. Zero issues there.  I have compiled it with the SX1280 modem and for the life of me cant figure this out. My OLED display works, TX LED is blinking and all. I am getting radio not found and can't get past this issue. I have double checked my pins, although the only thing I am not 100% on is DIO pin, I know on the 1276 for example this goes to DIO0 but the sx1280 does not have a DIO0 and has DIO1-3. I assumed DIO1 is the correct pin based off what I have read but I am not 100% sure.  Any help would be appreciated.

Thanks!

**Comment by btalbot85** [comment]
It seems something is going on with the pins I choose. I switched them around and I will get a firmware corrupt as well when using different pins and moving the sx1280 connections.  When disconnecting the miso/mosi or css pin then it goes back to radio not found.  I am not sure if it helps but I am using one of the lora1280 modules that look similar to the rfm95w modules. I have a feeling I am very close as the rnode boots up it looks like and when I run an rnodeconf --info I show validated locally.

**Comment by btalbot85** [comment]
Disregard, something was up with my build environment. Spent hours debugging when I simply needed a clean start ... live and learn I guess :)

**Reply by markqvist** [reply]
Great to hear you got it up and running!

Feel free to post a picture or two of your build if you feel like it! Would be very cool to see :)

**Reply by btalbot85** [reply]
Thanks! Currently I have moved it to a soldered prototype breadboard from a solderless breadboard. Now that I have confirmed functionality I have started working on the PCB design as well and will be moving away from just using a development board and module and plan to build this from the ground up. So far I have most of the code added and have made a board definition as well 
![PXL_20240430_070016308](https://github.com/markqvist/Reticulum/assets/79027771/96068859-7b94-4ab4-948c-241302eae969)

**Comment by markqvist** [comment]
Awesome, that is looking great! Thanks for sharing it!

Really looking forward to seeing what you come up with for the final design!

**Comment by tokio69** [comment]
Do you have a scematic or pinout on how you connected the ESP32 to the tranceiver? 
I have a simular setup with a ESP-32 S3 board and a E22-900M30S tranceiver but I have the "Radio Not Found" error on the OLED. Just like you, I double checked all the wires. I use the pre-compiled Heltec Lora32 v3 zipfile cause that is all the same hardware I thought, and I copied the scematics from Heltec pinouts and connections. But it seems the S3 is not "seeing" the SX1262. I left the RX and TX pins on the E22 as NC.
Pinout like this as now : 

GPIO 08 to NSS
GPIO 09 to SCK
GPIO 10 to MOSI
GPIO 11 to MISO
GPIO 12 to RST
GPIO 13 to BUSY
GPIO 14 to DIO1

RXEN, TXEN, DIO2 are not connected on the E22-900M30S.
OLED works on GPIO 17 and 18 with the radio not found error.
Maybe I also have to learn how to compile a custom firmware.
